packages/wdio-browserstack-service/src/util.ts

```diff
@@ -1359,3 +1480,68 @@ export const isValidCapsForHealing = (caps: { [key: string]: Options.Testrunner
     // Check if there are any capabilities and if at least one has a browser name
     return capValues.length > 0 && capValues.some(hasBrowserName)
 }
+
+type PollingResult = {
+    data: any;
+    headers: Record<string, any>;
+    message?: string; // Optional message for timeout cases
+  };
+
+async function pollApi(
```

Comment:
```
  async function customPollApi(  url,
    params,
    headers,
    upperLimit,
    startTime = Date.now()) {
      return new Promise( async (resolve, reject) => {
        while(true) {
          try {
            params.timestamp = Date.now() / 1000
            BStackLogger.debug(`current timestamp ${params.timestamp}`)
            const response = await axios.get(url, { params, headers })
      
            // If the request succeeds (non-404), return the result
            resolve({
                data: response.data,
                headers: response.headers,
                message: 'Polling succeeded.',
            });
        } catch (error) {
          if (axios.isAxiosError(error) && error.response?.status === 404) {
              const nextPollTime = parseInt(error.response.headers.next_poll_time, 10) * 1000
              BStackLogger.debug(`timeInMillis ${nextPollTime}`)
              if (isNaN(nextPollTime)) {
                  BStackLogger.warn('Invalid or missing `nextPollTime` header. Stopping polling.')
                  reject({
                      data: null,
                      headers: error.response.headers,
                      message: 'Invalid timeInMillis header value. Polling stopped.',
                  })
              }
    
              const elapsedTime = nextPollTime - Date.now()
              BStackLogger.debug(`elapsedTime ${elapsedTime} timeInMillis ${nextPollTime} upperLimit ${upperLimit}`)
              // Stop polling if the upper time limit is reached
              if (nextPollTime > upperLimit) {
                  BStackLogger.warn('Polling stopped due to upper time limit.')
                  resolve({
                      data: null,
                      headers: error.response.headers,
                      message: 'Polling stopped due to upper time limit.',
                  })
              }
    
              BStackLogger.debug(`Polling again in ${elapsedTime}ms with params:`, params)
    
              // Wait for the specified time and poll again
              await new Promise(resolve => setTimeout(resolve, elapsedTime))
          } else if (axios.isAxiosError(error)) {
              reject({ data: {}, headers: {}, message: error?.response?.data.message });
          } else {
            BStackLogger.error('Unexpected error occurred:')
            reject({ data: {}, headers: {}, message: 'Unexpected error occurred.' });
          }
        }
        }

      });
    }
```

Reply:
shouldn't it be something like this?

Reply:
I have handled it via recursive call and setTimeOut so the behaviour will be same